cmake/alpakaCommon.cmake

```diff
@@ -561,7 +561,7 @@ if(ALPAKA_ACC_GPU_HIP_ENABLE)
             set(ALPAKA_HIP_COMPILER "hipcc" CACHE STRING "HIP compiler")
             set_property(CACHE ALPAKA_HIP_COMPILER PROPERTY STRINGS "hipcc")
 
-            option(ALPAKA_HIP_FAST_MATH "Enable fast-math" ON)
+            option(ALPAKA_HIP_FAST_MATH "Enable fast-math" OFF)
```

Comment:
suggestion: Instead of hard set this option we should maybe use the compiler default. If we use cmake options with three states: `AUTO, OFF, ON` we could use `AUTO` as default.

Why I suggest it: IMO the goal of the alpaka's build system should be to set compiler options only if necessary e.g. workarounds or if the user explicit ask for it.
This would also solve all discussions about the default value.

Reply:
If we enable `AUTO` as an option, could we not remove the option entirely then? Users can always add the necessary flags themselves.

Reply:
It is possible, IMO not preferred, to remove it but these options are supported for the user. If you integrate alpaka into your project and you do not like to maintain a list of options per compilers, supporting it in alpaka is a multiplication so you do not add those options to any project.
I am also happy that we have those fast switches because I am jumping daily between different compilers and can not keep all options in my head.
An additional point is that it is not obvious how to provide these flags to the compiler because you need to know how alpaka is handling compile options internally.

Reply:
I also think that keeping the option is easier for users. I just wondered if we could remove some of our developer complexity. Anyway, let's keep the flag.

Reply:
Btw: I would not need the `AUTO` option. Having just a bool with `OFF` as default is fine for me!

Reply:
On the VC we decided to generally follow the `ON - OFF -AUTO` approach with cmake options. I made a new issue #1286 for it.